feat: add C implementation for `math/base/special/fast/uint32-sqrt`

Userf1995 opened the pull request:
Resolves #1932 

## Description

> What is the purpose of this pull request?

This pull request:

-   add C implementation for `math/base/special/fast/uint32-sqrt`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1932 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [X] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

Userf1995 committed:
feat: @stdlib/math/base/special/fast/uint32sqrt

Userf1995 committed:
Rename main.c to floor.c

Signed-off-by: Rejoan Sardar <119718513+Rejoan-Sardar@users.noreply.github.com>

Userf1995 committed:
Update benchmark.c

Signed-off-by: Rejoan Sardar <119718513+Rejoan-Sardar@users.noreply.github.com>

Userf1995 committed:
Update and rename floor.c to main.c

Signed-off-by: Rejoan Sardar <119718513+Rejoan-Sardar@users.noreply.github.com>

Userf1995 committed:
Rename main.c to floor.c

Signed-off-by: Rejoan Sardar <119718513+Rejoan-Sardar@users.noreply.github.com>

Userf1995 committed:
Update floor.c

Signed-off-by: Rejoan Sardar <119718513+Rejoan-Sardar@users.noreply.github.com>

Userf1995 committed:
Update floor.c

Signed-off-by: Rejoan Sardar <119718513+Rejoan-Sardar@users.noreply.github.com>

Userf1995 committed:
Update benchmark.c

Signed-off-by: Rejoan Sardar <119718513+Rejoan-Sardar@users.noreply.github.com>

Userf1995 committed:
Update and rename floor.c to main.c

Signed-off-by: Rejoan Sardar <119718513+Rejoan-Sardar@users.noreply.github.com>

Userf1995 committed:
Rename main.c to floor.c

Signed-off-by: Rejoan Sardar <119718513+Rejoan-Sardar@users.noreply.github.com>

Userf1995 committed:
Update floor.c

Signed-off-by: Rejoan Sardar <119718513+Rejoan-Sardar@users.noreply.github.com>

Userf1995 committed:
Update floor.c

Signed-off-by: Rejoan Sardar <119718513+Rejoan-Sardar@users.noreply.github.com>

Userf1995 committed:
Update floor.c

Signed-off-by: Rejoan Sardar <119718513+Rejoan-Sardar@users.noreply.github.com>

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/fast/uint32-sqrt/manifest.json:
```diff
@@ -0,0 +1,73 @@
+{
+	"options": {
```
Please update the indentation to two spaces in the `manifest.json` file.

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/fast/uint32-sqrt/src/floor.c:
```diff
@@ -0,0 +1,59 @@
+/**
+* Returns an integer square root.
+*
+* @param {uinteger32} x - input value
+* @returns {uinteger32} integer square root
```
```suggestion
* @param x    input value
* @return     integer square root
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/fast/uint32-sqrt/src/floor.c:
```diff
@@ -0,0 +1,59 @@
+/**
```
Shouldn't this file be named e.g. `sqrt.c`? 

Notice that we would need to rename the `src` fields in the `manifest.json` file as well.

Planeshifter reviewed (changes requested):
Thanks! Left a few comments and questions that need to be addressed, but otherwise this PR should be ready to land.

Userf1995 committed:
Update manifest.json

Signed-off-by: Rejoan Sardar <119718513+Rejoan-Sardar@users.noreply.github.com>

Userf1995 committed:
Update and rename floor.c to main.c

Signed-off-by: Rejoan Sardar <119718513+Rejoan-Sardar@users.noreply.github.com>

Userf1995 commented:
> Thanks! Left a few comments and questions that need to be addressed, but otherwise this PR should be ready to land.

@Planeshifter I have addressed all the issues you mentioned in your review.

Planeshifter reviewed (approved):
LGTM; thanks! Let's get it in.
